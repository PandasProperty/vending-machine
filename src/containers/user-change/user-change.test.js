import React from 'react';
import { Provider } from 'react-redux';
import UserChange from '.';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe('Vending Machine', () => {
    const store = mockStore({
        vendingMachine: {
            userChange: {
                split: [
                    {
                        id: '0',
                        label: '50 bani',
                        quantity: 10,
                        value: 0.5
                    },
                    {
                        id: '1',
                        label: '1 leu',
                        quantity: 0,
                        value: 0.5
                    }
                ],
                success: 10,
                value: 5
            }
        },
    });
    
    it('User-change componet', () => {
        const component = renderer.create(
            <Provider store={store}>
                <UserChange />
            </Provider>
        );
        
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
