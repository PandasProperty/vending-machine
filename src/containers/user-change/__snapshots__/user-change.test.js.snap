// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vending Machine User-change componet 1`] = `
<div
  className="user-change"
  onClick={[Function]}
>
  <div
    className="user-change__container"
  >
    <div
      className="user-change__label"
    >
      Your change: 5 RON
    </div>
    <div
      className="user-change__description"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-coins fa-w-16 fa-2x user-change__icon"
        data-icon="coins"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 405.3V448c0 35.3 86 64 192 64s192-28.7 192-64v-42.7C342.7 434.4 267.2 448 192 448S41.3 434.4 0 405.3zM320 128c106 0 192-28.7 192-64S426 0 320 0 128 28.7 128 64s86 64 192 64zM0 300.4V352c0 35.3 86 64 192 64s192-28.7 192-64v-51.6c-41.3 34-116.9 51.6-192 51.6S41.3 334.4 0 300.4zm416 11c57.3-11.1 96-31.7 96-55.4v-42.7c-23.2 16.4-57.3 27.6-96 34.5v63.6zM192 160C86 160 0 195.8 0 240s86 80 192 80 192-35.8 192-80-86-80-192-80zm219.3 56.3c60-10.8 100.7-32 100.7-56.3v-42.7c-35.5 25.1-96.5 38.6-160.7 41.8 29.5 14.3 51.2 33.5 60 57.2z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <div
        className="user-change__split"
      >
        <div
          className="user-change__coin"
        >
          10
           of 
          50 bani
        </div>
        <div
          className="user-change__coin"
        >
          0
           of 
          1 leu
        </div>
      </div>
    </div>
  </div>
</div>
`;
